Algebraic_kernel_d/examples/Algebraic_kernel_d/CMakeLists.txt

```diff
@@ -1,31 +1,60 @@
-# Created by the script cgal_create_cmake_script
-# This is the CMake script for compiling a CGAL application.
+# Created by the script cgal_create_CMakeLists
+# This is the CMake script for compiling a set of CGAL applications.
 
+project( Algebraic_kernel_d_examples )
 
-project( Algebraic_kernel_d_test )
 
 cmake_minimum_required(VERSION 2.8.11)
 
+# CGAL and its components
+find_package( CGAL QUIET COMPONENTS Core )
 
-find_package(CGAL QUIET COMPONENTS Core MPFI)
+if ( NOT CGAL_FOUND )
+  message(STATUS "This project requires the CGAL library, and will not be compiled.")
+  return()  
+endif()
 
-if ( CGAL_FOUND )
+# include helper file
+include( ${CGAL_USE_FILE} )
 
-  include( ${CGAL_USE_FILE} )
-  include( CGAL_CreateSingleSourceCGALProgram )
-  include( CGAL_VersionUtils )
 
-  include_directories (BEFORE ../../include)
+find_package( Boost REQUIRED )
+if ( NOT Boost_FOUND )
+  message(STATUS "This project requires the Boost library, and will not be compiled.")
+  return()  
+endif()
 
-  create_single_source_cgal_program( "Compare_1.cpp" )
-  create_single_source_cgal_program( "Construct_algebraic_real_1.cpp" )
-  create_single_source_cgal_program( "Isolate_1.cpp" )
-  create_single_source_cgal_program( "Sign_at_1.cpp" )
-  create_single_source_cgal_program( "Solve_1.cpp" )
 
-else()
-  
-    message(STATUS "This program requires the CGAL library, and will not be compiled.")
-  
+find_package( MPFI REQUIRED )
```

Comment:
Makes sense, I'll fix that. Does the same rationale apply to Boost, or is it that dependency special? I ask because lines 21-25 are generated exactly in that fashion by cgal_create_CMakeLists.